src/components/tableV2/hooks/useInfiniteScroll.js

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useMemo, useState } from "react"
+
+const useIniniteScroll = ({
+  service,
+  target,
+  onError,
+  onSuccess,
+  onDataend,
+  updateTableData,
+  threshold = 100,
+}) => {
+  const [lastReceivedData, setLastReceivedData] = useState()
+  const [loading, setLoading] = useState(false)
+
+  const hasDataEnded = useMemo(() => {
+    if (!onDataend) return false
+    return onDataend(lastReceivedData)
+  }, [lastReceivedData])
+
+  const runService = async () => {
+    try {
+      const currentData = await service(lastReceivedData)
+
+      updateTableData(currentData.data)
+      setLastReceivedData(currentData)
+      onSuccess?.(currentData)
+      setLoading(false)
+    } catch (err) {
+      onError?.(err)
+      setLoading(false)
+    }
+  }
+
+  const loadMore = () => {
```

Comment:
Both this, `scrollMethod ` and `runService` should be created inside the useEffect, no need to recreate on every rerender.